app/scripts/migrations/106.ts

```diff
@@ -0,0 +1,73 @@
+import { cloneDeep, isEmpty } from 'lodash';
+import {
+  ACCOUNT_LABEL_CHAIN_ID,
+  PertinentState,
+} from '../lib/setupAccountLabelsPetnamesBridge';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 106;
+
+/**
+ * Copy all account identity entries from PreferencesController to NameController.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, any>) {
+  const identities: PertinentState['identities'] =
+    state?.PreferencesController?.identities ?? {};
+  const names = state?.NameController?.names?.ethereumAddress ?? {};
+
+  if (isEmpty(Object.keys(identities))) {
+    return;
+  }
+
+  for (const address of Object.keys(identities)) {
+    const accountEntry = identities[address];
+
+    const normalizedAddress = address.toLowerCase();
+    const nameEntry = names[normalizedAddress] ?? {};
+    const petnameExists = Boolean(nameEntry[ACCOUNT_LABEL_CHAIN_ID]?.name);
```

Comment:
We'll likely have to use the default variation here so we create all the relevant petnames only once for existing account identities.

Reply:
Done.